### DISCUSSION TITLE
CGNS Boundary condition

### DISCUSSION THREAD

**Discussion by krupalbhat** [CGNS/CGNS#370] [CGNStalk]
Greetings,
I am trying to write cgns Boundary conditions using CGNS mid level library.
 I have created bc data set and a dirichlet BC data node.
Inside dirichlet BC Data I have created a data array of size 3 to write velocity x, y and z. But whenever I read it back it's showing "Wrong array size for Dirichlet data" . 
After further inspection I noticed that under dirichlet data node, all arrays should be of size 1 or of size of patch. 
Is there anyway to overcome this ??

I have gone through Sids document. I wanted to know more regarding CGNS Boundary condition. Is there any other references or examples which I can follow??

Also wanted to know if there is any software which can write cgns Boundary conditions data?

Thanks,

**Comment by mennodeij** [comment]
You probably can do this by writing separate 1-sized arrays for `VelocityX`, `VelocityY` and `VelocityZ`. This is also done for the vector field data and grid coordinates.

**Reply by krupalbhat** [reply]
Is it possible to write strings in Dirichlet node data array??

**Reply by awoimbee** [reply]
As far as I'm aware, CGNS only supports 1 dimensional arrays and every possible type is defined by DataType_t:
https://github.com/CGNS/CGNS/blob/b79637fcf5e83c7ea20fe017862ab4a97a886b65/src/cgnslib.h#L488
